test/integration/include/findAll.test.js

```diff
@@ -2054,4 +2054,53 @@ describe(Support.getTestDialectTeaser('Include'), () => {
         });
     });
   });
+
+  it.only('should be able to generate a correct request with inner and outer join', function() {
+    const Customer = this.sequelize.define('customer', {
+      name: DataTypes.STRING
+    });
+
+    const ShippingAddress = this.sequelize.define('shippingAddress', {
+      address: DataTypes.STRING,
+      verified: DataTypes.BOOLEAN
+    });
+
+    const Order = this.sequelize.define('purchaseOrder', {
+      description: DataTypes.TEXT
+    });
+
+    const Shipment = this.sequelize.define('shipment', {
+      trackingNumber: DataTypes.STRING
+    });
+
+    Customer.hasMany(ShippingAddress);
+    ShippingAddress.belongsTo(Customer);
+
+    Customer.hasMany(Order);
+    Order.belongsTo(Customer);
+
+    Shipment.belongsTo(Order);
+    Order.hasOne(Shipment);
+
+    return this.sequelize.sync({ force: true })
+    .then(() => {
+      return Shipment.findOne({
+        include: [{
+          model: Order,
+          required: true,
+          include: [{
+            model: Customer,
+            include: [{
+              model: ShippingAddress,
+              where: { verified: true }
+            }]
+          }]
+        }]
+      })
+      .then(result => {
+        //We just want to check that the query correctly runs
+        expect(result).to.not.be.ok;
```

Comment:
this assertion wouldn't fail if the query fails, so it doesn't seem to serve a purpose